### PR TITLE
HPCC-25402 Fix compiler issues with cassandraembed and eclagent on Ubuntu 20.10

### PR DESCRIPTION
**Pull request by Michael-Gardner**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 Please review. Found a second compiler issue with eclagent.cpp and ignore-qualifiers warnings as well as the cassandraembed deprecated-declarations warnings. Tested the docker build to ensure my changes didn't break anything, and there's a jenkins build available if you want to review the logs.

File: ecl/eclagent/eclagent.cpp [hunk 1]
```diff
@@ -3629,7 +3629,7 @@ extern int HTHOR_API eclagent_main(int argc, const char *argv[], StringBuffer *
                 }
 
                 const char * appName = argv[0];
-                for (int finger=0; argv[0][finger] != (const char)NULL; finger++)
+                for (int finger=0; argv[0][finger] != (char)NULL; finger++)
```
Comment (richardkchapman) [inline ecl/eclagent/eclagent.cpp:3632]: Should be '\0' rather than (char) NULL

**Comment by xwang2713** [comment]
Agree with Richard's comment

File: ecl/eclagent/eclagent.cpp (hunk 1 above)
Comment (richardkchapman) [reply ecl/eclagent/eclagent.cpp:3632]: Looks like Jake already fixed that issue in a separate PR...

**Comment by richardkchapman** [comment]
@Michael-Gardner Please rebase for a clean merge

File: ecl/eclagent/eclagent.cpp (hunk 1 above)
Comment (Michael-Gardner) [reply ecl/eclagent/eclagent.cpp:3632]: Removed the commit that touches eclagent.cpp since Jake fixed it.

**Comment by Michael-Gardner** [comment]
Removed the conflict that Jake fixed in a separate PR. Rebased to master to include his change and I'm re-testing. I'll @ you when it's complete.

**Comment by Michael-Gardner** [comment]
@richardkchapman rebased and retested on jenkins. 20.10 is now building fine.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**